### PR TITLE
Upgrade TypeScript to 5.0

### PR DESCRIPTION
**Pull request by nickgros**

### CODE REVIEW

File: shared/tsconfig.base.json [hunk 1]
```diff
@@ -22,6 +22,7 @@
     "skipLibCheck": true,
     "strictNullChecks": true,
     "suppressImplicitAnyIndexErrors": true,
+    "ignoreDeprecations": "5.0",
```
Comment (nickgros) [inline shared/tsconfig.base.json:25]: suppressImplicitAnyIndexErrors is deprecated and will be removed in TS5.5. We will need to refine our types in some places to remove this flag.

Comment (nickgros) [reply shared/tsconfig.base.json:25]: https://sagebionetworks.jira.com/browse/PORTALS-2598

File: apps/synapse-oauth-signin/package.json [hunk 2]
```diff
@@ -66,7 +66,7 @@
     "markdown-it-synapse-math": "^3.0.4",
     "markdown-it-synapse-table": "^1.0.6",
     "memfs": "^3.4.7",
-    "msw": "1.0.1",
```
Comment (nickgros) [inline apps/synapse-oauth-signin/package.json:69]: Upgrading MSW past v1 in SRC seems to break a few things. Let's use the same version (0.49.3) across the monorepo

**Review by jay-hodgson — APPROVED** [review]

**Review by bleu4422 — APPROVED** [review]
[]()

**Pull request merged**